### PR TITLE
Escape quotes in value for radio matcher

### PR DESCRIPTION
**Pull request by jeremywadsack**
Under jQuery 1.12, the provided expression for matching a radio button when the value contains a quote throws a "Syntax error, unrecognized expression".

This resolves that by properly escaping quote characters in the value before building the matching expression.

Note that in order to prove this and build a failing test I fixed up `test/index.html` to pull in `qunit` from the `vendor` folder (it was missing). This is because the phantomjs runner is broken right now in the `npm`/`karma` tests. (There's some discussion of this under #307.)

Then I found that this doesn't fail in the bundled jQuery (1.8.3), but does fail with 1.12.x that we are using. I updated the reference in `test/index.html` to use the version from the jQuery CDN just to make the test show that the code is fixed. I don't know if you want to update the vendored jQuery to the latest from the 1.x branch, but I'm happy to make that change as well.

### CODE REVIEW

File: test/index.html [hunk 1]
```diff
@@ -3,18 +3,19 @@
...
+  <meta charset="UTF-8">
 
-	<link rel="stylesheet" href="vendor/qunit.css" type="text/css"/>
+	<link rel="stylesheet" href="vendor/qunit-2.0.1.css" type="text/css"/>
 
 	<!--script type="text/javascript" src="http://cdnjs.cloudflare.com/ajax/libs/zepto/1.1.1/zepto.min.js"></script>
 	<script type="text/javascript" src="vendor/zepto-data.js"></script-->
-	<script type="text/javascript" src="vendor/jquery.js"></script>
-	<script type="text/javascript" src="vendor/qunit.js"></script>
+	<script type="text/javascript" src="https://code.jquery.com/jquery-1.12.4.min.js"></script>
```
Comment (akre54) [inline test/index.html]: Can you please keep this as the vendored code so we don't need the internet to run the tests?

File: test/index.html [hunk 2]
```diff
@@ -3,18 +3,19 @@
...
+	<script type="text/javascript" src="vendor/qunit-2.0.1.js"></script>
```
Comment (akre54) [inline test/index.html]: Please split this change into another PR. And replace the file in place instead of creating a new one?

File: backbone.stickit.js [hunk 3]
```diff
@@ -449,7 +449,7 @@
     selector: 'input[type="radio"]',
     events: ['change'],
     update: function($el, val) {
-      $el.filter('[value="'+val+'"]').prop('checked', true);
+      $el.filter('[value="'+val.toString().replace(/"/g, '\\"')+'"]').prop('checked', true);
```
Comment (akre54) [inline backbone.stickit.js]: Please pull this variable out to the line above. Perhaps with a comment explaining what it is for.

File: test/index.html (hunk 2 above)
Comment (jeremywadsack) [inline test/index.html]: Done. See #311.

File: backbone.stickit.js (hunk 3 above)
Comment (jeremywadsack) [inline backbone.stickit.js]: Good point.

**Comment by jeremywadsack** [comment]
@akre54 Fixes done. Rebased, squashed, and pushed. 

Let me know if there are any other changes that I should look at.

Thanks for the quick response on this!

File: backbone.stickit.js [hunk 4]
```diff
@@ -449,7 +449,11 @@
     selector: 'input[type="radio"]',
     events: ['change'],
     update: function($el, val) {
-      $el.filter('[value="'+val+'"]').prop('checked', true);
+      // Because we use " in the value descriptor, we need to escape any
+      // quotes in the value itself otherwise jQuery/Sizzle complain that
+      // the selector is invalid.
+      escapedVal = val.toString().replace(/"/g, '\\"')
```
Comment (akre54) [inline backbone.stickit.js]: Needs a `var` before `escapedVal`

Comment (jeremywadsack) [inline backbone.stickit.js]: Ack, good catch. Too much coffee script lately.

Fixed, sqaushed, pushed.
